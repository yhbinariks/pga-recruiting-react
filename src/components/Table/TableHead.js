import React from 'react';
import PropTypes from 'prop-types';

const TableHead = ({ setSortBy }) => (
  <thead>
    <tr>
      <th onClick={() => setSortBy('name')}>Name</th>
      <th onClick={() => setSortBy('score')}>Score</th>
      <th />
    </tr>
  </thead>
);

const propTypes = {
  setSortBy: PropTypes.func.isRequired,
};

TableHead.propTypes = propTypes;

export default TableHead;
