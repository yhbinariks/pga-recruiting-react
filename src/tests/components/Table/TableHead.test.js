import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow, mount } from 'enzyme';
import TableHead from '../../../components/Table/TableHead';

describe('TableHead component', () => {
  it('snapshot', () => {
    const props = {
      setSortBy: jest.fn(),
    };
    const tableHead = shallow(
      <table>
        <TableHead {...props} />
      </table>,
    );
    expect(toJson(tableHead)).toMatchSnapshot();
  });
  it('sortByName', () => {
    const props = {
      setSortBy: jest.fn(),
    };
    const tableHead = mount(
      <table>
        <TableHead {...props} />
      </table>,
    );
    tableHead.find('th').at(0).simulate('click');
    expect(props.setSortBy).toHaveBeenCalledWith('name');
  });
  it('sortByScore', () => {
    const props = {
      setSortBy: jest.fn(),
    };
    const tableHead = mount(
      <table>
        <TableHead {...props} />
      </table>,
    );
    tableHead.find('th').at(1).simulate('click');
    expect(props.setSortBy).toHaveBeenCalledWith('score');
  });
});
