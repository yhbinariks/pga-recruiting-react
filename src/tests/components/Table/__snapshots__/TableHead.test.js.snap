// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableHead component snapshot 1`] = `
<table>
  <TableHead
    setSortBy={[MockFunction]}
  />
</table>
`;
